docs-internal/multipart.md

```diff
@@ -0,0 +1,328 @@
+    Json.createObjectBuilder().add("foo", "bar").build(),
+    Json.createObjectBuilder().add("alice", "bob").build());
+response.send(mp);
+```
+
+### Process a request content with the reactive model
+
+The following example consumes each part content as a `JsonObject`:
+
+```java
+req.content().as(MultiPart.class).thenAccept((multiPart) -> {
+    multiPart.subscribe(new Flow.Subscriber<BodyPart>(){
+        @Override
+        public void onSubscribe(Flow.Subscription subscription) {
+            subscription.request(Long.MAX_VALUE);
+        }
+
+        @Override
+        public void onNext(BodyPart bodyPart) {
+            bodyPart.content().as(JsonObject.class).thenAccept((str) -> {
+                System.out.println("File uploaded: " + bodyPart.headers().filename());
+            });
+        }
+
+        @Override
+        public void onError(Throwable throwable) {
+            res.status(500);
+            res.send();
+        }
+
+        @Override
+        public void onComplete() {
+            System.out.println("sending response");
+            res.send("Files uploaded successfully");
+        }
+    });
+});
+```
+
+The following example consumes each part content using reactive stream subscribers, `ServerFileWriter` is inspired from the streaming example at `webserver/examples/streaming`:
+
+```java
+req.content().as(MultiPart.class).thenAccept((multiPart) -> {
+    multiPart.subscribe(new Flow.Subscriber<BodyPart>(){
+        @Override
+        public void onSubscribe(Flow.Subscription subscription) {
+            subscription.request(Long.MAX_VALUE);
+        }
+
+        @Override
+        public void onNext(BodyPart bodyPart) {
+            bodyPart.content().subscribe(new ServerFileWriter());
+        }
+
+        @Override
+        public void onError(Throwable throwable) {
+            res.status(500);
+            res.send();
+        }
+
+        @Override
+        public void onComplete() {
+            System.out.println("sending response");
+            res.send("Files uploaded successfully");
+        }
+    });
+});
+```
+
+### Return a response with the reactive model
+
+The following example returns a multipart response using publishers of `Datachunk`, `ServerFileReader` is inspired from the streaming example at `webserver/examples/streaming`:
+
+```java
+MultiPart mp = MultiPart.builder()
+    .bodyPart(BodyPart.create(
+        BodyPartHeaders.builder().
+            .contentType(MediaType.APPLICATION_JSON)
+            .filename("file1")
+            .build(),
+        new ServerFileReader(filePath1))
+    .bodyPart(BodyPart.create(
+        BodyPartHeaders.builder().
+            .contentType(MediaType.APPLICATION_JSON)
+            .filename("file2")
+            .build(),
+        new ServerFileReader(filePath2))
+    .build();
+response.send(mp);
+```
+
+## API
+
+```java
+public interface BodyPart {
+
+    ServerRequest request();
+
+    ServerResponse response();
+
```

Comment:
I did add that as a convenience there is no hard requirement.
Agreed that this can removed and if the user wants a reference to the request and response this is something they can do themselves.

Reply:
I've removed the methods, marking as resolved.